<link rel="import" href="../search/comparators.html">
<script src="../search/query.js"></script>

<dom-module id="sugar-task-search-input">
  <template>
    <style>
      #queryParseError {
        color: red;
      }
    </style>
    <input value="{{queryString::input}}" />
    <div id="queryParseError">[[queryParseError]]</div>
  </template>
  <script>
Polymer({
  is: 'sugar-task-search-input',

  properties: {
    queryParseError: String,
    queryString: String,
    search: {
      type: Object,
      computed: 'computeSearch_(queryString)',
      notify: true
    }
  },

  computeSearch_: function(queryString) {
    try {
      var search = query.parse(queryString);
      this.set('queryParseError', '');
      return search;
    } catch(e) {
      // TODO(charliea): It seems like what we do on failure here has big
      // ramifications, but I haven't really though them through yet.
      this.set('queryParseError', e.message);
    }

    return undefined;
  }
});
</script>
</dom-module>
